File: fole_backend/src/main/java/findr/fole/model/Room.java
package findr.fole.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "room")
public class Room {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String name;
    @ManyToOne
    private RoomType roomType;
    @ManyToOne
    private Building building;
}

File: frontend/fole-findr-app/src/app/model/room.ts
import {RoomType} from "./room-type";
import {BuildingFloor} from "./building-floor";

export class Room {
  id!: number;
  name!: string;
  roomType!: RoomType;
  roomDescr!: string;
  buildingFloorDTO!: BuildingFloor;
}
